### PR TITLE
Enable address sanitizer in tsfile cpp.

### PR DESCRIPTION
**Pull request by ColinLeeo**
In this pr:
1. Fixed the issue where a flush operation would result in an E_ALLREAY_EXIST error after inserting a large amount of data.
2. Enabled Address Sanitizer in tests.
3. Checked and fixed memory leaks.
4. Removed some unused configuration items.
5. Performed code formatting and cleanup.

### CODE REVIEW

File: cpp/src/common/tsfile_common.h [hunk 1]
```diff
@@ -580,6 +581,13 @@ class TSMIterator {
     common::SimpleList<ChunkGroupMeta *> &chunk_group_meta_list_;
     common::SimpleList<ChunkGroupMeta *>::Iterator chunk_group_meta_iter_;
     common::SimpleList<ChunkMeta *>::Iterator chunk_meta_iter_;
+    std::map<common::String, std::map<common::String, std::vector<ChunkMeta *>>>
+        tsm_info_;
+    std::map<common::String,
+             std::map<common::String, std::vector<ChunkMeta *>>>::iterator
+        tsm_device_iter_;
+    std::map<common::String, std::vector<ChunkMeta *>>::iterator
+        tsm_measurement_iter_;
```
Comment (jt2594838) [inline cpp/src/common/tsfile_common.h]: Add some comments to explain what each String indicates or use better names.

**Review by jt2594838 — APPROVED** [review]

**Comment by ColinLeeo** [comment]
It seems that compiling tsfile_cpp with asan at Windows needs clang instead of Mingw.

**Comment by jt2594838** [comment]
> It seems that compiling tsfile_cpp with asan at Windows needs clang instead of Mingw.

Yeah, how about only enabling it on linux. I think it enough for detecting memory leaks.

File: cpp/src/reader/aligned_chunk_reader.cc [hunk 2]
```diff
@@ -513,6 +513,8 @@ int AlignedChunkReader::decode_time_value_buf_into_tsblock(
         if (!prev_time_page_not_finish()) {
             time_in_.reset();
         }
+        value_page_col_notnull_bitmap_.clear();
```
Comment (ColinLeeo) [inline cpp/src/reader/aligned_chunk_reader.cc:516]: Frequent resizing of vectors can lead to high memory usage and may cause AddressSanitizer (ASan) to report errors.

File: cpp/test/file/open_file_test.cc [hunk 3]
```diff
@@ -40,11 +40,6 @@ class OpenFileTest : public ::testing::Test {
     OpenFile* open_file_;
 };
 
-TEST_F(OpenFileTest, Initialization) {
```
Comment (ColinLeeo) [inline cpp/test/file/open_file_test.cc:43]: This unit test is unnecessary and can cause memory leaks during testing.

File: cpp/src/common/datatype/value.h [hunk 4]
```diff
@@ -33,16 +33,41 @@
...
+                break;
+            case TEXT:
+                value_.sval_ = nullptr;
+                break;
+            case NULL_TYPE:
+                break;
+            default:
+                LOGE("unknown data type");
+        }
+    }
```
Comment (ColinLeeo) [inline cpp/src/common/datatype/value.h:61]: Initializing a union by setting only a single variable to 0 may not be sufficient, as the variable might not be long enough, causing other types to remain non-zero.

File: cpp/examples/cpp_examples/demo_write.cpp [hunk 5]
```diff
@@ -20,39 +20,97 @@
...
-    int row_count = 100;
-    for (int i = 1; i < row_count; ++i) {
-        storage::DataPoint point(measurement_name, 10000 + i);
-        storage::TsRecord record(i, device_name, 1);
-        record.points_.push_back(point);
-        ret = tsfile_writer.write_record(record);
-        ASSERT(ret == 0);
+    TsFileWriter* tsfile_writer_ = new TsFileWriter();
+    libtsfile_init();
+    std::cout << "input tablesize" << std::endl;
```
Comment (jt2594838) [inline cpp/examples/cpp_examples/demo_write.cpp]: This line should be moved to the line before line 85, namely 'std::cin >> table_size;'.
And beware of the terms "Table" and "Tablet".

**Review by jt2594838 — APPROVED** [review]

**Review by jt2594838 — APPROVED** [review]

**Pull request merged**